plugin/src/main/java/io/jenkins/plugins/casc/plugins/PluginManagerConfigurator.java

```diff
@@ -1,416 +0,0 @@
-import java.io.IOException;
-import java.io.InputStream;
-import java.io.PrintWriter;
-import java.net.URI;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.LinkedList;
-import java.util.List;
-import java.util.Map;
-import java.util.Queue;
-import java.util.Set;
-import java.util.concurrent.ExecutionException;
-import java.util.concurrent.TimeUnit;
-import java.util.jar.JarFile;
-import java.util.logging.Level;
-import java.util.logging.Logger;
-import java.util.stream.Collectors;
-
-import static java.nio.charset.StandardCharsets.UTF_8;
-
-/**
- * @author <a href="mailto:nicolas.deloof@gmail.com">Nicolas De Loof</a>
- */
-@Extension(ordinal = 999)
-@Restricted(NoExternalUse.class)
-public class PluginManagerConfigurator extends BaseConfigurator<PluginManager> implements RootElementConfigurator<PluginManager> {
-
-    private final static Logger logger = Logger.getLogger(PluginManagerConfigurator.class.getName());
-
-    @Override
-    public Class<PluginManager> getTarget() {
-        return PluginManager.class;
-    }
-
-    @Override
-    public PluginManager getTargetComponent(ConfigurationContext context) {
-        return Jenkins.getInstance().getPluginManager();
-    }
-
-    @Override
-    protected PluginManager instance(Mapping mapping, ConfigurationContext context) {
-        return getTargetComponent(context);
-    }
-
-    @Override
-    protected void configure(Mapping config, PluginManager instance, boolean dryrun, ConfigurationContext context) throws ConfiguratorException {
-
-        // PluginManager has no dry-run mode : we need to actually install plugins, or we just can't check
-        // the other elements of the configuration regarding required plugins.
-
-        Mapping map = config.asMapping();
-        final Jenkins jenkins = Jenkins.getInstance();
-
-        configureProxy(map, jenkins, context);
-
-        final UpdateCenter updateCenter = configureUpdateSites(map, jenkins, context);
-
-        configurePlugins(map, jenkins, updateCenter, context);
-
-        try {
-            jenkins.save();
-        } catch (IOException e) {
-            throw new ConfiguratorException("failed to save Jenkins configuration", e);
-        }
-    }
-
-    private void configureProxy(Mapping map, Jenkins jenkins, ConfigurationContext context) throws ConfiguratorException {
-        final CNode proxy = map.get("proxy");
-        if (proxy != null) {
-            Configurator<ProxyConfiguration> pc = context.lookup(ProxyConfiguration.class);
-            if (pc == null) throw new ConfiguratorException("ProxyConfiguration not well registered");
-            ProxyConfiguration pcc = pc.configure(proxy, context);
-            jenkins.proxy = pcc;
-        }
-    }
-
-    private UpdateCenter configureUpdateSites(Mapping map, Jenkins jenkins, ConfigurationContext context) throws ConfiguratorException {
-        final CNode sites = map.get("sites");
```

Comment:
Update site require some more code it seems 😄

Reply:
ah damn